src/client.js

```diff
@@ -0,0 +1,58 @@
+require('babel-polyfill')
+const uuid = require('uuid/v4')
+const defaultConfig = require('./config/default.json')
+const logger = require('./lib/logger')
+const producer = require('./producer')
+
+class Client {
+  constructor (config, topic) {
+    this.config = config || defaultConfig
+    this.topic = topic || this.config.kafkatopic
+    if (!this.topic) {
+      throw Error('no kafka topic informed')
+    }
+    this.logger = logger.child({
+      source: 'eventsgateway/client',
+      topic: this.topic,
+    })
+    const pClass = this.config.producer.async ? producer.Async : producer.Sync
+    this.producer = new pClass(this.config, this.logger)
+  }
+
+  sendToTopic (name, topic, props) {
+    if (!name) {
+      throw Error('event name cannot be empty')
+    }
+    if (!topic) {
+      throw Error('topic cannot be empty')
+    }
+    // currently there are no interceptors in node-grpc
+    // the proposal was already accepted
+    // https://github.com/grpc/proposal/blob/master/L5-NODEJS-CLIENT-INTERCEPTORS.md
+    // but the PR is still ongoing
+    // https://github.com/grpc/grpc-node/pull/59
```

Comment:
It looks like the PR was already merged.

Reply:
I checked this yesterday but I wasn't sold on the idea that we gain a lot by changing how it already is done. What are your points about using an interceptor vs what we have right now?

Reply:
I don't have a strong opinion actually. Using the interceptor is a bit cleaner but my main issue here is that: 
1. The comment is no longer valid since the PR is merged.
2. It is in the wrong place. We would use the interceptor in `this.grpcClient` and not when building the event, right?

If we choose to move on without the interceptor I'd just remove this comment to avoid future confusion.

Reply:
alright! removing the comment